### PR TITLE
Accept Pandas Series in lov property

### LINKED ISSUE #1444: Accept Pandas Series in lov property
### Description

A user had an issue putting a Pandas Series as a Lov property, as it is not allowed. A better warning has been issued since the 3.1 release. Accepting a Pandas series as a potential value for the Lov property would be great.

```python
from taipy.gui import Gui
import pandas as pd

data = pd.DataFrame({"Countries":["France", "Germany", "Italy", "Spain"],
                     "Population": [65, 83, 59, 47]})
selected = data["Countries"][0]
selection = data["Countries"]

page = "<|{selected}|selector|lov={selection}|dropdown|>"

Gui(page).run(port="auto")

```

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Propagate any change on the demos and run all of them to ensure there is no breaking change.
- [ ] Ensure any change is well documented.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by Satoshi-Sh**
## Related Issue 
Fixes #1444

## Updates 
- Add a condition to convert a pandas.Series to a list in the `_adapter.py`
- Add a new test case to `test_selector.py`

## Comments 
I made a new test case for pd.Series but it doesn't pass. Can you have a look at it?

Screenshot from 2024-06-25 05-19-15

### CODE REVIEW

**Comment by FlorianJacta** [comment]
This issue is, in fact, still being discussed with the R&D. We will get back to you when the R&D agrees

**Comment by Satoshi-Sh** [comment]
Got it. Is it better to close this or fine to keep this as is? @FlorianJacta

**Comment by FlorianJacta** [comment]
Let's keep it for the moment

File: taipy/gui/utils/_adapter.py [hunk 1]
```diff
@@ -60,6 +62,8 @@ def _get_unique_type(self, type_name: str) -> str:
 
     def run(self, var_name: str, value: t.Any, id_only=False) -> t.Any:
         lov = _AdaptedLov.get_lov(value)
+        if isinstance(lov, pd.Series):
```
Comment (FredLL-Avaiga) [inline taipy/gui/utils/_adapter.py]: this 2 lines could go into the _AdaptedLov.get method

Comment (Satoshi-Sh) [reply taipy/gui/utils/_adapter.py]: Thanks for checking, @FredLL-Avaiga 

Has the RD agreed on this feature?

Comment (FredLL-Avaiga) [reply taipy/gui/utils/_adapter.py]: I think it's more a bug fix than a feature, so you can go ahead

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
simple and efficient
Could you update your branch ?

**Comment by Satoshi-Sh** [comment]
It's updated but the test I wrote fails for some reason. Can you have a look at it?

File: tests/gui/builder/control/test_selector.py [hunk 2]
```diff
@@ -69,3 +71,19 @@ def test_selector_builder_3(gui: Gui, test_client, helpers):
...
     ]
     helpers.test_control_builder(gui, page, expected_list)
+
+
+def test_selector_builder_4(gui: Gui, test_client, helpers):
+    pd_series = pd.Series(["l1", "l2", "l3"])
+    gui._bind_var_val("selected_val", "l1")
+    gui._bind_var_val("selector_properties", pd_series)
+    with tgb.Page(frame=None) as page:
+        tgb.selector(value="{selected_val}", lov="{selector_properties}")
```
Comment (FredLL-Avaiga) [inline tests/gui/builder/control/test_selector.py]: need to add `# type: ignore[attr-defined]` to avoid the linter error

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
you also need to manage the type in builder.py (for the default value) in `def __get_list_of()`
I would recommend a generic approach

```
if not isinstance(lof, list) and hasattr(lof, "tolist"):
            try:
                return lof.tolist()  #type: ignore[union-attr]
            except Exception as e:
                _warn("Error accessing List of values", e)
```

File: tests/gui/builder/control/test_selector.py (hunk 2 above)
Comment (Satoshi-Sh) [reply tests/gui/builder/control/test_selector.py]: Thanks for your help. The test passes once updated the `builder.py`

**Review by FredLL-Avaiga — APPROVED** [review]

**Comment by FredLL-Avaiga** [comment]
If you update your branch and resolve the conversations, I think we can merge this branch.

**Comment by Satoshi-Sh** [comment]
@FredLL-Avaiga  Resolved conversations and updated the branch.

**Review by FredLL-Avaiga — DISMISSED** [review]
I'd like @FabienLelaquais and @dinhlongviolin1 to take a look

**Comment by FabienLelaquais** [comment]
> d with the R&D. We will get back to you when the R&D agrees

On it, @FredLL-Avaiga

File: tests/gui/builder/control/test_selector.py [hunk 3]
```diff
@@ -69,3 +71,19 @@ def test_selector_builder_3(gui: Gui, test_client, helpers):
         "value={_TpLv_tpec_TpExPr_selected_obj_TPMDL_0}",
     ]
     helpers.test_control_builder(gui, page, expected_list)
+
+
+def test_selector_builder_4(gui: Gui, test_client, helpers):
```
Comment (FabienLelaquais) [inline tests/gui/builder/control/test_selector.py]: I would have renamed this test to `test_selector_pandas_series`, which really is the point of it.

**Review by FabienLelaquais — DISMISSED** [review]
This is really simple and clear.
I added a comment just in case. Do it or not. Let me know.
No problem to approve+merge then.

File: tests/gui/builder/control/test_selector.py (hunk 3 above)
Comment (Satoshi-Sh) [reply tests/gui/builder/control/test_selector.py]: The function name sounds much better. I updated it.

**Review by FredLL-Avaiga — APPROVED** [review]

**Review by FabienLelaquais — APPROVED** [review]
👌

**Comment by FredLL-Avaiga** [comment]
Thank you @Satoshi-Sh

**Comment by Satoshi-Sh** [comment]
Thanks for your help, Fred :rocket:

**Pull request merged**